src/coord/src/catalog/storage.rs

```diff
@@ -438,6 +440,29 @@ impl Transaction<'_> {
         }
     }
 
+    fn validate_timestamp_bindings(&self, source_id: &GlobalId) -> Result<(), String> {
+        let bindings_vec = self
+            .load_timestamp_bindings(*source_id)
+            .map_err(|e| format!("{}", e))?;
+
+        let bindings_by_pid = bindings_vec.iter().group_by(|(pid, _ts, _offset)| pid);
+
+        for (pid, bindings) in &bindings_by_pid {
+            let mut latest_offset = 0;
+            for (_pid, _ts, offset) in bindings {
```

Comment:
you might as well assert that the timestamps go up too? I know `load` does an order by, but that way you'll catch bugs in `load` too in the unlikely event that it ever happens?

Reply:
:+1: